File: exam_portal/exam-portal/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root',
})
export class QuizService {
  constructor(private _http: HttpClient) {}
  public quizzes() {
    return this._http.get(`${baseUrl}/quiz/`);
  }

  public addQuiz(quizData: any) {
    return this._http.post(`${baseUrl}/quiz/`, quizData);
  }
  public deleteQuiz(qId: any) {
    return this._http.delete(`${baseUrl}/quiz/${qId}`);
  }
  public getQuiz(qId: any) {
    return this._http.get(`${baseUrl}/quiz/${qId}`);
  }
  public updQuiz(quizData: any) {
    return this._http.put(`${baseUrl}/quiz/`, quizData);
  }
  // get quizzes of category
  public getQuizzesOfCategory(cId: any) {
    return this._http.get(`${baseUrl}/quiz/category/${cId}`);
  }
  // Get active quizzes
  public getActiveQuizzes(){
    return this._http.get(`${baseUrl}/quiz/active/`);
  }
   // Get active quizzes of category
   public getActiveQuizzesOfCategory(cId:any){
    return this._http.get(`${baseUrl}/quiz/category/active/${cId}`);
  }
}

File: Exam Portal/src/main/java/com/example/service/QuizService.java
package com.example.service;

import com.example.model.exam.Category;
import com.example.model.exam.Quiz;

import java.util.List;
import java.util.Set;

public interface QuizService {
    Quiz addQuiz(Quiz quiz);

    Quiz updateQuiz(Quiz quiz);

    Set<Quiz> getQuizzes();

    Quiz getQuiz(Long qId);

    void deleteQuid(Long qId);


    List<Quiz> getQuizzesByCategoryId(Category category);
    public  List<Quiz> getActiveQuizzes();
    public List<Quiz> getActiveQuizzesOfCategory( Category category);
}
